update hello world tutorial (#97)
aiken-lang/site: src/pages/example--hello-world/end-to-end/mesh.mdx

First, we setup Mesh with Blockfrost as a provider. This will allow us to let Mesh handle transaction building for us, which includes managing changes. It also gives us a direct way to submit the transaction later on.

Create a file named `common.ts` in the root of your project and add the following code:

```ts filename="common.ts"
import fs from "node:fs";
import {
  BlockfrostProvider,
  MeshTxBuilder,
  MeshWallet,
  serializePlutusScript,
  UTxO,
} from "@meshsdk/core";
import { applyParamsToScript } from "@meshsdk/core-csl";

> // BlockfrostProvider to query data from the blockchain
> const blockchainProvider = new BlockfrostProvider(
>   fs.readFileSync("bf.key").toString()
> );

Reviewer: ```suggestion
const blockchainProvider = new BlockfrostProvider(process.env.BLOCKFROST_PROJECT_ID);
```

Let's keep the ENV var here, thanks 🙏

// wallet for signing transactions
export const wallet = new MeshWallet({
  networkId: 0,
  fetcher: blockchainProvider,
  submitter: blockchainProvider,
  key: {
    type: "root",
    bech32: fs.readFileSync("me.sk").toString(),
  },
});
```

> Note that the highlighted line above looks Blockfrost project ID, so we create a file named `bf.key` and add your Blockfrost project ID to it.

Reviewer: ```suggestion
<Callout type="warning">
Note that the highlighted line above looks for an environment variable named `BLOCKFROST_PROJECT_ID` which its value must be set to your Blockfrost project id.
You can define a new environment variable in your terminal by running (in the same session you're also executing the script!):
```

> ```console filename="bf.key"
> preview...

Reviewer: ```suggestion
export BLOCKFROST_PROJECT_ID=preview...
```

```

Replace `preview...` with your actual project id.

> 

Reviewer: ```suggestion
</Callout>

```

Next, we'll need to read the validator from the blueprint (`plutus.json`) we generated earlier. We'll also need to convert it to a format that Mesh understands. This is done by serializing the validator and then converting it to a hexadecimal text string, we can do this by using the `applyParamsToScript` function.

```ts filename="common.ts" {27-42}
import fs from "node:fs";
import {
  BlockfrostProvider,
  MeshTxBuilder,
  MeshWallet,
  serializePlutusScript,
  UTxO,
} from "@meshsdk/core";
import { applyParamsToScript } from "@meshsdk/core-csl";

> // BlockfrostProvider to query data from the blockchain
> const blockchainProvider = new BlockfrostProvider(
>   fs.readFileSync("bf.key").toString()
> );

Reviewer: ```suggestion
const blockchainProvider = new BlockfrostProvider(process.env.BLOCKFROST_PROJECT_ID);
```

// wallet for signing transactions
export const wallet = new MeshWallet({
  networkId: 0,
  fetcher: blockchainProvider,
  submitter: blockchainProvider,
  key: {
    type: "root",
    bech32: fs.readFileSync("me.sk").toString(),
  },
});

// script information, its CBOR and address
import blueprint from "./plutus.json";

export function getScript() {
  const scriptCbor = applyParamsToScript(
    blueprint.validators[0].compiledCode,
    []
  );

  const scriptAddr = serializePlutusScript(
    { code: scriptCbor, version: "V3" },
  ).address;

  return { scriptCbor, scriptAddr };
}
```

Lastly, let's add 2 more useful functions to our `common.ts` file. One to get a transaction builder and another function to fetch a UTxO by transaction hash.

```ts filename="common.ts" {43-58}
import fs from "node:fs";
import {
  BlockfrostProvider,
  MeshTxBuilder,
  MeshWallet,
  serializePlutusScript,
  UTxO,
} from "@meshsdk/core";
import { applyParamsToScript } from "@meshsdk/core-csl";

> // BlockfrostProvider to query data from the blockchain
> const blockchainProvider = new BlockfrostProvider(
>   fs.readFileSync("bf.key").toString()
> );

Reviewer: ```suggestion
const blockchainProvider = new BlockfrostProvider(process.env.BLOCKFROST_PROJECT_ID);
```

    .txInCollateral(
      collateral.input.txHash,
      collateral.input.outputIndex,
      collateral.output.amount,
      collateral.output.address
    )
    .selectUtxosFrom(utxos)
    .complete();
  const unsignedTx = txBuilder.txHex;

  const signedTx = await wallet.signTx(unsignedTx);
  const txHash = await wallet.submitTx(signedTx);
  console.log(`1 tADA unlocked from the contract at Tx ID: ${txHash}`);
}

main();
```

<Callout type="info">
Run this script as usual, but this time, also passing the transaction id obtained from the previous command locking the funds. For example:

```console
npx tsx unlock.ts 48b8178e3a8842227dfbb0f73669efc163f73fd7c8758b7dafc0a5a5f07a5445
```
</Callout>

If everything worked as planned you should see something resembling the following output:

```
1 tADA unlocked from the contract at Tx ID: af3e7d83e5ee5324a612de9126636ef55505ffb9c468dacaf419635921b7a91c
```

And, tada 🎉!

We can inspect our [redeeming transaction on CardanoScan](https://preview.cardanoscan.io/transaction/d3d5e828a3989691b0960d22a265c8c9ae4723134b52aa05ec0fb7d40f060392?tab=contracts) and see that it successfully executed our _Hello World_ contract.

> You can find the [full source code](https://github.com/MeshJS/examples/tree/main/aiken-hello-world) for this tutorial on GitHub.

Reviewer: ```suggestion
You can find the [full source code](https://github.com/MeshJS/examples/tree/main/aiken-hello-world) for this tutorial on MeshJS's GitHub.
```
